tests/test_base.py

```diff
@@ -636,3 +655,189 @@ def test_pack_header_for_user(self, app, user_class):
             assert token_data['rls'] == 'admin,operator'
             assert token_data['duder'] == 'brief'
             assert token_data['el_duderino'] == 'not brief'
+
+    def test_registration_email(self, app, user_class, db):
+        """
+        This test verifies that Praetorian encrypts passwords using the scheme
+        specified by the HASH_SCHEME setting. If no scheme is supplied, the
+        test verifies that the default scheme is used. Otherwise, the test
+        verifies that the encrypted password matches the supplied scheme.
+        """
+        stock_config = app.config.copy()
+
+        _pwd = dirname(dirname(abspath(__file__)))
+        here = _pwd + '/flask_praetorian/templates'
+        tmpl_file = here + '/registration_email.html'
+        with open(tmpl_file) as _template:
+            tmpl = Template(_template.read())
+
+        # generate (w/o sending) a notification email and match it to our own
+        app.config['TESTING'] = True
+        app.config['PRAETORIAN_EMAIL_TEMPLATE'] = tmpl_file
+        app.config['PRAETORIAN_CONFIRMATION_ENDPOINT'] = 'unprotected'
+
+        default_guard = Praetorian(app, user_class)
+
+        # create our default test user
+        the_dude = user_class(
+            username='TheDude',
+            email='the@dude.com',
+            password=default_guard.hash_password('abides'),
+        )
+        db.session.add(the_dude)
```

Comment:
probably better to flush and rollback at the end of the test. If there's a failure in the middle, we'll have residual data in the database